n2v/methods/oucarter.py

```diff
@@ -0,0 +1,351 @@
+
+            tauL_rho_DFT_beta = tauLmP_rho_DFT_beta + tauP_rho_DFT_beta
+            tauL_rho_beta = tauLmP_rho_beta + tauP_rho_beta
+
+            vext_opt_no_H_DFT_beta = e_bar_DFT_beta - tauL_rho_DFT_beta - vxc_LDA_DFT_beta
+            vext_opt_no_H_beta = e_bar_beta - tauL_rho_beta - vxc_LDA_beta
+
+            vext_opt_no_H_DFT_Fock_beta = self.dft_grid_to_fock(vext_opt_no_H_DFT_beta, self.Vpot)
+            vext_opt_DFT_Fock_beta = vext_opt_no_H_DFT_Fock_beta - J[0] - J[1]
+
+            vext_opt_beta = vext_opt_no_H_beta - vH
+
+            # vext_opt_DFT_Fock = (vext_opt_DFT_Fock + vext_opt_DFT_Fock_beta) * 0.5
+            # vext_opt = (vext_opt + vext_opt_beta) * 0.5
+
+            return (vext_opt_DFT_Fock, vext_opt_DFT_Fock_beta), (vext_opt, vext_opt_beta)
+        return vext_opt_DFT_Fock, vext_opt
+    
+    def oucarter(self, maxiter, vxc_grid, D_tol=1e-7,
+             eig_tol=1e-4, frac_old=0.5, init="scan"):
+        """
+        (23) in [1].
+        [1] [J. Chem. Theory Comput. 2018, 14, 5680−5689]
+
+                parameters:
+        ----------------------
+            maxiter: int
+                same as opt_max_iter
+            vxc_grid: np.ndarray of shape (3, num_grid_points)
+                The final result will be represented on this grid
+                default: 1e-4
+            D_tol: float, opt
+                convergence criteria for density matrices.
+                default: 1e-7
+            eig_tol: float, opt
+                convergence criteria for occupied eigenvalue spectrum.
+                default: 1e-4
+            frac_old: float, opt
+                Linear mixing parameter for current vxc and old vxc.
+                If 0, no old vxc is mixed in.
+                Should be in [0,1)
+                default: 0.5.
+            init: string, opt
+                Initial guess method.
+                default: "SCAN"
+                1) If None, input wfn info will be used as initial guess.
+                2) If "continue" is given, then it will not initialize
+                but use the densities and orbitals stored. Meaningly,
+                one can run a quick WY calculation as the initial
+                guess. This can also be used to user speficified
+                initial guess by setting Da, Coca, eigvec_a.
+                3) If it's not continue, it would be expecting a
+                method name string that works for psi4. A separate psi4 calculation
+                would be performed.
+        """
+        
+        # if self.ref != 1:
+        #     raise ValueError("Currently only supports Spin-Restricted "
+        #                      "calculations since Spin-Unrestricted CI "
+        #                      "is not supported by Psi4.")
+
+        
+        if self.guide_potential_components[0] != "hartree":
+            raise ValueError("Hartree potential is necessary as the guide potential.")
```

Comment:
Suggestion:
`if "hartree" not in self.guide_potential_components:`

Unless you want to halt if anything other than hartree is used... Which is probably what we want right? Not sure if LDA would mess up with everything since its added to the other components.

Reply:
Yeah make sense.

Reply:
Will yes. I took a look and realized that Inverse.invert should do the correction if not hartree and here was designed to halt. LDA will mess it up.